src/test.js

```diff
@@ -537,18 +544,31 @@ function checkPollution() {
 
 // Will be exposed as QUnit.asyncTest
 function asyncTest( testName, expected, callback ) {
+	var warning;
+
+	if ( global.console && global.console.warn ) {
+		warning = "DEPRECATED: QUnit.asyncTest() will be removed in QUnit 2.0.\n" +
+			"Reference: https://api.qunitjs.com/QUnit.asyncTest/";
+		global.console.warn(warning);
```

Comment:
the console.warn call needs to be abstracted. Otherwise you'll need to check for it every time you trigger a warning.

Also, the messages could be stored on a dictionary object, easier to keep.